### PR TITLE
doc: similarity search tutorial

### PR DESCRIPTION
**Pull request by jiashenC**
1. Add a sift feature extractor for getting image feature.
2. Add jupyter-notebook for similarity search tutorial.

### CODE REVIEW

File: eva/udfs/sift_feature_extractor.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import cv2
+import kornia
+import numpy as np
+import pandas as pd
+import torch
+
+from eva.udfs.abstract.abstract_udf import AbstractClassifierUDF
+
+
+class SiftFeatureExtractor(AbstractClassifierUDF):
```
Comment (gaurav274) [inline eva/udfs/sift_feature_extractor.py:26]: Please use decorators for the UDF. Moving forward, we want to completely shift to decorators.

File: eva/udfs/udf_bootstrap_queries.py [hunk 2]
```diff
@@ -150,6 +150,15 @@
     EVA_INSTALLATION_DIR
 )
 
+Sift_udf_query = """CREATE UDF IF NOT EXISTS SiftFeatureExtractor
+        INPUT  (frame NDARRAY UINT8(3, ANYDIM, ANYDIM))
```
Comment (gaurav274) [inline eva/udfs/udf_bootstrap_queries.py:154]: You can skip INPUT/OUTPUT
Change the TYPE to FeatureExtraction

File: eva/udfs/sift_feature_extractor.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import pandas as pd
+import torch
+
+from eva.catalog.catalog_type import NdArrayType
+from eva.udfs.abstract.abstract_udf import AbstractUDF
+from eva.udfs.decorators.decorators import forward, setup
+from eva.udfs.decorators.io_descriptors.data_types import PandasDataframe
+
+
+class SiftFeatureExtractor(AbstractUDF):
```
Comment (gaurav274) [inline eva/udfs/sift_feature_extractor.py:27]: Also inherit `GPUCompatible` so that it can run on GPU.

Comment (gaurav274) [reply eva/udfs/sift_feature_extractor.py:27]: You also need to implement `to_device()` in that case

File: tutorials/09-similarity-sesarch-for-motif-mining.ipynb [hunk 4]
```diff
@@ -0,0 +1,724 @@
+{
```
Comment (gaurav274) [inline tutorials/09-similarity-sesarch-for-motif-mining.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    print(response)
</pre><p>Change it to response.as_df()</p><p>same for other cellls</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/georgia-tech-db/eva/pull/700/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (gaurav274) [inline tutorials/09-similarity-sesarch-for-motif-mining.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    """
</pre><p>Convert these cells to markdown for readibility</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/georgia-tech-db/eva/pull/700/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (gaurav274) [inline tutorials/09-similarity-sesarch-for-motif-mining.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    import matplotlib.pyplot as plt
</pre><p>Should we advertise a visually better example?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/georgia-tech-db/eva/pull/700/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

**Review by jiashenC — COMMENTED** [review]
The notebook looks good to me.

**Pull request merged**